hackathon_site/applications/models.py

```diff
@@ -0,0 +1,46 @@
+from django.db import models
+from django.contrib.auth.models import User
+
+import uuid
+
+
+def _generate_team_code():
+    team_code = uuid.uuid4().hex[:5].upper()
+    while Team.objects.filter(team_code=team_code).exists():
+        team_code = uuid.uuid4().hex[:5].upper()
+    return team_code
+
+
+class Team(models.Model):
+    team_code = models.CharField(max_length=5, default=_generate_team_code, null=False)
+
+    created_at = models.DateTimeField(auto_now_add=True, null=False)
+    updated_at = models.DateTimeField(auto_now=True, null=False)
+
+    MAX_MEMBERS = 4
+
+    def __str__(self):
+        return self.team_code
+
+
+class Application(models.Model):
+    user = models.ForeignKey(User, on_delete=models.CASCADE, null=False)
+    team = models.ForeignKey(Team, on_delete=models.CASCADE, null=False)
+
+    # User Submitted Fields
```

Comment:
Missing ethnicity, t-shirt size, dietary restrictions, school, program, graduation year, "how did you hear about makeuoft", 2 more checkboxes

Reply:
This is meant to be a placeholder / template, not for MakeUofT specific - I think it's fine as is with the most basic of fields, specific events can add the ones they want in when this repo is forked